gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/sideEffect/GroupSideEffectStep.java

```diff
@@ -81,6 +81,14 @@ public void modulateBy(final Traversal.Admin<?, ?> kvTraversal) {
     }
 
     @Override
+    public void replaceLocalChild(final Traversal.Admin<?, ?> oldTraversal, final Traversal.Admin<?, ?> newTraversal) {
+        if (null != this.keyTraversal && this.keyTraversal.equals(oldTraversal))
+            this.keyTraversal = this.integrateChild(newTraversal);
+        else if (null != this.valueTraversal && this.valueTraversal.equals(oldTraversal))
```

Comment:
Like `GroupStep`, there is some pre-processing required on the `valueTraversal`. I would add a two new test cases to `GroupTest` and `GroupSideEffectTest` which verify proper functioning. For instance -- `...group().by(label()).by(label().count())...` (and then the same traversal if you want, using `group('m')`.